spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala

```diff
@@ -1,91 +1,121 @@
 import filodb.core.{DatasetRef, GlobalConfig}
 import filodb.core.store.ScanSplit
+import filodb.downsampler.chunk.DownsamplerSettings
+
 
 /**
   * Contains all the objects necessary for performing worker tasks. Is constructed from SparkConf
   * and is cached.
   */
 class ChunkCopier(conf: SparkConf) {
   private def openConfig(str: String) = {
-    val sysConfig = GlobalConfig.systemConfig.getConfig("filodb")
-    ConfigFactory.parseFile(new File(conf.get(str))).getConfig("filodb").withFallback(sysConfig)
+    ConfigFactory.parseFile(new File(conf.get(str))).withFallback(GlobalConfig.systemConfig)
   }
 
-  // Both "source" and "target" refer to file paths which define config files that have a
-  // top-level "filodb" section and a "cassandra" subsection.
-  val sourceConfig = openConfig("spark.filodb.chunkcopier.source.configFile")
-  val targetConfig = openConfig("spark.filodb.chunkcopier.target.configFile")
+  def datasetConfig(mainConfig: Config, datasetName: String): Config = {
+    def getConfig(path: lang.String): Config = {
+      ConfigFactory.parseFile(new File(path))
+    }
 
+    val sourceConfigPaths: util.List[lang.String] = mainConfig.getStringList("dataset-configs")
+    sourceConfigPaths.stream()
+      .map[Config](new util.function.Function[lang.String, Config]() {
+        override def apply(path: lang.String): Config = getConfig(path)
+      })
+      .filter(new util.function.Predicate[Config] {
+        override def test(conf: Config): Boolean = conf.getString("dataset").equals(datasetName)
+      })
+      .findFirst()
+      .orElseThrow()
+  }
+
+  val rawSourceConfig = openConfig("spark.filodb.chunks.copier.source.configFile")
+  val rawTargetConfig = openConfig("spark.filodb.chunks.copier.target.configFile")
+  val sourceConfig = rawSourceConfig.getConfig("filodb")
+  val targetConfig = rawTargetConfig.getConfig("filodb")
   val sourceCassConfig = sourceConfig.getConfig("cassandra")
   val targetCassConfig = targetConfig.getConfig("cassandra")
 
-  val sourceDatasetRef = DatasetRef.fromDotString(conf.get("spark.filodb.chunkcopier.source.dataset"))
-  val targetDatasetRef = DatasetRef.fromDotString(conf.get("spark.filodb.chunkcopier.target.dataset"))
+  val datasetName = conf.get("spark.filodb.chunks.copier.dataset")
+  val datasetRef = DatasetRef.fromDotString(datasetName)
+  val targetDatasetConfig = datasetConfig(targetConfig, datasetName)
 
   // Examples: 2019-10-20T12:34:56Z  or  2019-10-20T12:34:56-08:00
   private def parseDateTime(str: String) = Instant.from(DateTimeFormatter.ISO_OFFSET_DATE_TIME.parse(str))
 
-  val ingestionTimeStart = parseDateTime(conf.get("spark.filodb.chunkcopier.ingestionTimeStart"))
-  val ingestionTimeEnd = parseDateTime(conf.get("spark.filodb.chunkcopier.ingestionTimeEnd"))
+  val ingestionTimeStart = parseDateTime(conf.get("spark.filodb.chunks.copier.repairStartTime"))
+  val ingestionTimeEnd = parseDateTime(conf.get("spark.filodb.chunks.copier.repairEndTime"))
```

Comment:
It will be helpful to retain the name "ingestion" in the config to remove ambiguity on whether it is user time or ingestion time. We may remember now, but can tend to forget